src/nwb_datajoint/common/common_filter.py

```diff
@@ -164,10 +164,10 @@ def filter_data_nwb(self, analysis_file_abs_path, eseries, filter_coeff, valid_t
         output_shape_list[electrode_axis] = len(electrode_ids)
         output_offsets = [0]
 
-        timestamp_size = sys.getsizeof(timestamps_on_disk[0])
-        timestamp_dtype = type(timestamps_on_disk[0])
-        data_size = sys.getsizeof(data_on_disk[0][0])
-        data_dtype = type(data_on_disk[0][0])
+        timestamp_size = timestamps_on_disk[0].itemsize
+        timestamp_dtype = timestamps_on_disk[0].dtype
+        data_size = data_on_disk[0][0].itemsize
```

Comment:
Same here. I also don't understand the `[0][0]` part. And I worry that converting it to numpy will cause the dataset to be loaded to memory, which we may not want to do for large datasets.

Reply:
You're right that this loads data, but only the first element, not the full data array. That's a general feature of h5py: you can pull slices of data into RAM without loading the full dataset. And as mentioned above, these calls do produce the right values (I checked).